mitodl/mitxonline
# Add hubspot_sync celery task queue specifically for hubspot related tasks

# What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/1720

# Description (What does it do?)
Creates a new hubspot specific celery task queue called `hubspot_sync`.  This queue will be used for any celery task with a namespace of `hubspot_sync.tasks.*`.  

Most of the HubSpot related tasks are very labor intensive and can degrade application performance.  By creating a separate celery task queue specifically for hubspot tasks, we can more easily kill the tasks if they begin to impact application performance.

# How can this be tested?

1. Checkout this PR branch.
2. Run `docker-compose build`
3. Run `docker-compose up`
4. Run `./manage.py sync_db_to_hubspot --users update`
5. Verify the celery container logs show that the hubspot task was received and successfully completed.

# Additional Context
Similar PR: https://github.com/mitodl/micromasters/commit/84747d1cca9bd5d894751fd05d17246d85e3f739#diff-0a99231995da379e7aebabe76c9d849a23737a42c3b3a8994043e2aa80958424

Celery documentation: https://docs.celeryq.dev/en/latest/userguide/routing.html#automatic-routing
